### PR TITLE
[MAJOR] Added end to end and backend tests to Task Management.

### PR DESCRIPTION
**Pull request by bowsong**
Reformat code and add e2e and repository tests

### CODE REVIEW

**Review by nasaku898 — DISMISSED** [review]

File: backend/src/main/dto/TaskDTOs.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ export interface TaskCreationDTO {
   deadlineDate: Date;
   createdDate: Date;
   modifiedDate: Date;
+  //To update in database: make projectId optional
```
Comment (Manraj45) [inline backend/src/main/dto/TaskDTOs.ts:8]: remove comment and create an issue about it

File: backend/src/main/services/AssignedService.ts [hunk 2]
```diff
@@ -33,7 +41,17 @@ export class AssignedService {
     const existingAssignments = await this.assignedRepository.getByTaskId(parseInt(taskId, 10));
     const emailList: string[] = [];
     const newEmailList: string[] = assignedToUpdateDTO.emails;
-    existingAssignments?.forEach((ass: Assigned) => {
+    if (assignedToUpdateDTO.taskId === undefined) {
+      throw new HttpException(StatusCodes.BAD_REQUEST, 'Request data is missing some values');
+    }
+
```
Comment (Manraj45) [inline backend/src/main/services/AssignedService.ts:47]: Spacing here is not consistent

File: backend/src/main/services/AssignedService.ts [hunk 3]
```diff
@@ -54,4 +72,23 @@ export class AssignedService {
...
     return this.assignedRepository.deleteById(parseInt(taskId, 10));
   };
+
+  public static isThereNullAssignedCreationDTO = (assignedCreationDTO: AssignedCreationDTO): boolean => {
+    if (assignedCreationDTO === undefined || !assignedCreationDTO.taskId || !assignedCreationDTO.email) {
+      return true;
+    }
+    return false;
+  };
+  public static isThereNullMultipleAssignedCreationDTO = (
```
Comment (Manraj45) [inline backend/src/main/services/AssignedService.ts:82]: You should add a space between functions

File: backend/src/main/services/TaskService.ts [hunk 4]
```diff
@@ -30,4 +35,19 @@ export class TaskService {
   public updateTask = async (taskId: string, taskUpdateDTO: TaskUpdateDTO): Promise<number> => {
     return this.taskRepository.update(parseInt(taskId, 10), taskUpdateDTO);
   };
+  public static isThereNullValueTaskCreationDTO = (taskCreationDTO: TaskCreationDTO): boolean => {
```
Comment (Manraj45) [inline backend/src/main/services/TaskService.ts:38]: same comment here. Should add a space before this function

File: backend/src/test/services/AssignedService.test.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+import { sequelizeMock } from '../helpers/SequelizeMock';
+import AssignedRepository from '../../main/repositories/AssignedRepository';
+import { container } from 'tsyringe';
+import { MultipleAssignedCreationDTO } from '../../main/dto/AssignedDTOs';
+import { Assigned } from '../../main/models/Assigned';
+import { AssignedService } from '../../main/services/AssignedService';
+
+describe('AssignedService tests', () => {
+  let assignedRepositoryMock: any = null;
+  beforeAll(() => {
```
Comment (Manraj45) [inline backend/src/test/services/AssignedService.test.ts:12]: Same here add space before functions. This comment applies for all the test files

File: backend/src/test/services/TaskService.test.ts [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+      createdDate: date,
+      modifiedDate: date,
+      projectId: 1
+    };
+    const taskService = container.resolve(TaskService);
+    await expect(taskService.createTask(NEW_TASK as TaskCreationDTO)).rejects.toThrowError(
+      'Request data is missing some values'
+    );
+  });
+  it('should fail because of missing value in request data (missing status)', async () => {
```
Comment (Manraj45) [inline backend/src/test/services/TaskService.test.ts:70]: Same here and for the next few "it"

File: frontend/cypress/integration/Task_Management/createTask.spec.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
+/// <reference types="cypress" />
+
+beforeEach(() => {
+  cy.visit('/');
+});
+it('Should create a new Task', () => {
```
Comment (Manraj45) [inline frontend/cypress/integration/Task_Management/createTask.spec.js:6]: Same comment as before for the spacing

File: frontend/src/components/Task/CreateTask.tsx [hunk 8]
```diff
@@ -118,15 +129,14 @@ const ManageTask: React.FC<any> = ({ id, edit }) => {
...
           const employees = await getAssignedByTaskId(id);
-          const taskData = taskToEdit.data;
+          const taskData: Task = taskToEdit.data;
           let tempEmployees: string[] = [];
 
           employees.data.forEach((element: any) => {
             tempEmployees.push(element.email);
           });
-
           //mega efficient (use moment.js or smt for dates)
```
Comment (Manraj45) [inline frontend/src/components/Task/CreateTask.tsx:138]: remove comments and console.logs if there are any

**Review by Manraj45 — CHANGES_REQUESTED** [review]
Good job! Need a few changes for the spacing between functions and removing unnecessary comments.

File: backend/src/main/dto/TaskDTOs.ts (hunk 1 above)
Comment (bowsong) [reply backend/src/main/dto/TaskDTOs.ts:8]: https://github.com/kevin-ghannoum/soen490/issues/112

**Review by nasaku898 — APPROVED** [review]

**Review by Manraj45 — APPROVED** [review]
Good Job!

**Pull request merged**